service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt

```diff
@@ -732,6 +734,30 @@ WHERE employee_id = :userId
         )
     }
 
+    fun requirePermissionFor(user: AuthenticatedUser, action: Action.VasuDocumentFollowup, id: VasuDocumentFollowupEntryId) {
+        if (action != Action.VasuDocumentFollowup.UPDATE_FOLLOWUP_ENTRY) {
+            throw Forbidden()
+        }
+
+        val globalRoles = when (user) {
+            is AuthenticatedUser.Employee -> user.globalRoles
+            else -> user.roles
+        }
```

Comment:
I think that `globalRoles` never includes `UNIT_SUPERVISOR`, `SPECIAL_EDUCATION_TEACHER` or `GROUP_STAFF`. You would have to use `acl.getRolesForChild` to get these scoped roles for a specific child.

Reply:
And that can obviously be done only after reading the vasu document from database to check which child "owns" it.